pandas/io/formats/style_render.py

```diff
@@ -477,6 +489,109 @@ def _generate_col_header_row(self, iter: tuple, max_cols: int, col_lengths: dict
 
         return index_blanks + column_name + column_headers
 
+    def _generate_descriptor_row(self, iter: tuple, max_cols: int):
+        """
+        Generate the row containing calculated descriptor values for columns:
+
+         +----------------------------+---------------+---------------------------+
+         |  index_blanks ...          | descriptor_i  |  value_i by col           |
+         +----------------------------+---------------+---------------------------+
+
+        Parameters
+        ----------
+        iter : tuple
+            Looping variables from outer scope
+        max_cols : int
+            Permissible number of columns
+
+        Returns
+        -------
+        list of elements
+        """
+
+        r, descriptor = iter
+
+        # number of index blanks is governed by number of hidden index levels
+        index_blanks = [
+            _element("th", self.css["blank"], self.css["blank_value"], True)
+        ] * (self.index.nlevels - sum(self.hide_index_) - 1)
+
+        if isinstance(descriptor, str):
+            name: str | None = descriptor
+            func: Callable = getattr(Series, descriptor)
+        elif isinstance(descriptor, tuple):
+            name, func = descriptor[0], descriptor[1]
+        else:
+            name, func = None, descriptor
+
+        display_func: Callable = _maybe_wrap_formatter(
+            formatter=None,  # use _default_formatter
+            decimal=get_option("styler.format.decimal"),
+            thousands=get_option("styler.format.thousands"),
+            precision=get_option("styler.format.precision"),
+            na_rep=get_option("styler.format.na_rep"),
+            escape=get_option("styler.format.escape"),
+        )
+
+        base_css = f"{self.css['descriptor_name']} {self.css['descriptor']}{r}"
+        descriptor_name = [
+            _element(
+                "th",
+                base_css
+                if (name is not None and not self.hide_column_names)
```

Comment:
This is awfully hard to read code with 2 ternary operators inside a function inside a list.

Reply:
You could defer the list packing until the end.

Reply:
- [x] chged